samwise.js

```diff
@@ -1,17 +1,17 @@
 /**
```

Comment:
SCRIPT-This function seems to have grown quite complex. Can we break it down further?